Doc/library/csv.rst

```diff
@@ -295,6 +295,11 @@ The :mod:`csv` module defines the following classes:
       Twenty rows after the first row are sampled; if more than half of columns +
       rows meet the criteria, :const:`True` is returned.
 
+      Additionally, if all columns are found to be strings and have varying
+      lengths, the average length of all the strings becomes a crucial factor
+      in the determination process.
```

Comment:
I found this rather vague, and would really recommend being specific here about _how_ the average length is used, and under what conditions it means this method returns ``True``, just like the rest of this description does for the other cases. Even skimming the code and description here it wasn't totally clear to me, so I didn't suggest something specific, but this should presumably look something like the following:

```rst
      lengths, the average length of each row is (used to/compared with) ...
      and if (greater than/less than) ... , ``True`` is returned.
```